molgenis-questionnaires/src/main/java/org/molgenis/questionnaires/service/impl/QuestionnaireServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+import static org.molgenis.security.core.utils.SecurityUtils.getCurrentUsername;
+
+@Service
+public class QuestionnaireServiceImpl implements QuestionnaireService
+{
+	private static final String DEFAULT_SUBMISSION_TEXT = "<h3>Thank you for submitting the questionnaire.</h3>";
+
+	private final DataService dataService;
+	private final EntityManager entityManager;
+	private final UserPermissionEvaluator userPermissionEvaluator;
+	private final QuestionnaireFactory questionnaireFactory;
+	private final StaticContentService staticContentService;
+
+	public QuestionnaireServiceImpl(DataService dataService, EntityManager entityManager,
+			UserPermissionEvaluator userPermissionEvaluator, QuestionnaireFactory questionnaireFactory,
+			StaticContentService staticContentService)
+	{
+		this.dataService = Objects.requireNonNull(dataService);
+		this.entityManager = requireNonNull(entityManager);
+		this.userPermissionEvaluator = requireNonNull(userPermissionEvaluator);
+		this.questionnaireFactory = requireNonNull(questionnaireFactory);
+		this.staticContentService = requireNonNull(staticContentService);
+	}
+
+	@Override
+	public List<QuestionnaireResponse> getQuestionnaires()
+	{
+		return dataService.query(ENTITY_TYPE_META_DATA, EntityType.class)
+						  .eq(EntityTypeMetadata.EXTENDS, QUESTIONNAIRE)
+						  .findAll()
+						  .filter(entityType -> userPermissionEvaluator.hasPermission(
+								  new EntityTypeIdentity(entityType.getId()), EntityTypePermission.WRITE))
+						  .map(entityType ->
+						  {
+							  String entityTypeId = entityType.getId();
+							  Questionnaire questionnaireEntity = findQuestionnaireEntity(entityTypeId);
+
+							  if (questionnaireEntity == null)
+							  {
+								  // Create a questionnaire entity for the current user
+								  Questionnaire questionnaire = questionnaireFactory.create(
+										  entityManager.create(entityType, POPULATE));
+
+								  questionnaire.setOwner(getCurrentUsername());
+								  questionnaire.setStatus(NOT_STARTED);
+								  dataService.add(entityType.getId(), questionnaire);
+								  return QuestionnaireResponse.create(questionnaire);
+							  }
+							  return QuestionnaireResponse.create(questionnaireEntity);
+						  })
+						  .collect(toList());
+	}
+
+	@Override
+	public void startQuestionnaire(String id)
+	{
+		Questionnaire questionnaire = findQuestionnaireEntity(id);
+		if (questionnaire.getStatus().equals(NOT_STARTED))
+		{
+			// Set questionnaire status to open once it has been requested
+			questionnaire.setStatus(OPEN);
+			dataService.update(id, questionnaire);
+		}
+	}
+
+	@Override
+	public String getQuestionnaireSubmissionText(String id)
+	{
+		String key = id + "_submissionText";
+		String submissionText = staticContentService.getContent(key);
+
+		if (submissionText == null)
+		{
+			submissionText = DEFAULT_SUBMISSION_TEXT;
+			staticContentService.submitContent(key, submissionText);
+		}
+
+		return submissionText;
```

Comment:
if submissionText is null you are returning null. this is a really bad idea, a it pushed the problem to the user of this method ( who does not know about this)